### PR TITLE
Added Review Api

### PR DESCRIPTION
**Pull request by ghost**
The core Magento 2 does not provide the REST API for the Reviews Extention of the products. So we have written Api for reviews and rating of a product.

We are attaching Documentation file based on Swagger .
[swaggerapi.txt](https://github.com/magento/magento2/files/331774/swaggerapi.txt)

### CODE REVIEW

**Comment by ghost** [comment]
What would be the next steps as our pull request has been passed all the checks should we close this request ?
Would you merge the code ?

**Comment by ghost** [comment]
@yaronish any updates on this ?

**Comment by sshrewz** [comment]
Internal ticket - MAGETWO-56790

File: app/code/Magento/Review/Api/ReviewInterface.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+     * @param string $title
+     * @param string $nickname
+     * @param string $detail
+     * @param int $ratingValue
+     * @param int $customerId
+     * @param int $storeId
+     * @return boolean
+     *
+     */
+    public function writeReviews(
```
Comment (cisco87) [inline app/code/Magento/Review/Api/ReviewInterface.php]: I think these are a lot of arguments for a method, is there anyway we can use parameter object or inject dependencies that can retrieve these values or maybe have ReviewInterface have these values as instance properties

File: app/code/Magento/Review/Model/ReviewManagement.php [hunk 2]
```diff
@@ -0,0 +1,235 @@
+<?php
+namespace Magento\Review\Model;
+
+use Magento\Review\Api\ReviewInterface;
+use Magento\Review\Model\Review;
+use Magento\Review\Model\ResourceModel\Review\CollectionFactory;
+
+class ReviewManagement implements \Magento\Review\Api\ReviewInterface
```
Comment (cisco87) [inline app/code/Magento/Review/Model/ReviewManagement.php]: This is being imported but you're using the full namespace here anyway, can you change it to ReviewInterface

**Comment by shaibar** [comment]
Hi @ajay-ipragmatech  your api is not working, i tested it on magento 2.1 and the swagger fails to create to generate the api.
I found the reason for it is that you specify a return value of array in the api. according to what i found magento 2.1 does not allow returning array you must define the type of array when returning it in the API
https://alankent.me/2015/07/24/creating-a-new-rest-web-service-in-magento-2/
> @alankent  - For rest API bindings we must be told the exact type. You cannot use “array”. You must use something like CustomerInterface[].

**Comment by okorshenko** [comment]
@ajay-ipragmatech thank you for your contribution. Unfortunately we can not accept this Pull Request in current implementation since it does not cover existing functionality. Rating object is more complex than int value. Please, check Product documentation for rating attributes: http://docs.magento.com/m2/ce/user_guide/stores/attributes-rating.html 

If you would like to implement Review WebAPI, please, let us know and we will share with you high-level design document for this functionality. 

Closing this Pull Request for now.

**Comment by okorshenko** [comment]
Related PR: https://github.com/magento/magento2/pull/4486

**Pull request closed without merging**